gateway-service/src/main/java/org/zowe/apiml/gateway/security/service/zosmf/ZosmfService.java

```diff
@@ -221,15 +220,23 @@ public String getZosmfRealm(String infoURIEndpoint) {
     /**
      * Verify whether the service is actually accessible.
      *
+     * Note: This method uses getURI, it's also verifying eureka registration
+     *
      * @return true when it's possible to access the Info endpoint via GET.
      */
     public boolean isAccessible() {
         final HttpHeaders headers = new HttpHeaders();
         headers.add(ZOSMF_CSRF_HEADER, "");
 
-        String infoURIEndpoint = getURI(getZosmfServiceId()) + ZOSMF_INFO_END_POINT;
-        log.debug("Verifying z/OSMF accessibility on info endpoint: {}", infoURIEndpoint);
+        String infoURIEndpoint = "";
+        try {
+            infoURIEndpoint = getURI(getZosmfServiceId()) + ZOSMF_INFO_END_POINT;
```

Comment:
What about using a URL builder (to mitigate clashing clashes)?

Reply:
Yes, it would be a good improvement

Reply:
done